tensorflow_lite_support/python/task/audio/core/tensor_audio.py

```diff
@@ -0,0 +1,158 @@
+    # Need to import the module to catch this error:
+    # `from pybind11_abseil import status`
+    # see https://github.com/pybind/pybind11_abseil#abslstatusor.
+    audio = _LoadAudioBufferFromFile(
+      file_name, sample_count, np.zeros([sample_count]))
+    tensor = TensorAudio(audio.audio_format, audio.buffer_size)
+    tensor.load_from_array(np.array(audio.float_buffer, copy=False))
+    return tensor
+
+  def load_from_audio_record(self, record: audio_record.AudioRecord) -> None:
+    """Loads audio data from an AudioRecord instance.
+    Args:
+      record: An AudioRecord instance.
+    Raises:
+      ValueError: Raised if the audio record's config is invalid.
+    """
+    if record.buffer_size < self._buffer_size:
+      raise ValueError(
+        "The audio record's buffer size cannot be smaller than the tensor "
+        "audio's sample count.")
+
+    if record.channels != self._format.channels:
+      raise ValueError(
+        f"The audio record's channel count doesn't match. "
+        f"Expects {self._format.channels} channel(s).")
+
+    if record.sampling_rate != self._format.sample_rate:
+      raise ValueError(
+        f"The audio record's sampling rate doesn't match. "
+        f"Expects {self._format.sample_rate}Hz.")
+
+    # Load audio data from the AudioRecord instance.
+    data = record.read(self._buffer_size)
+    self.load_from_array(data.astype(np.float32))
+
+  def load_from_array(self,
+                      src: np.ndarray,
+                      offset: int = None,
+                      size: int = None) -> None:
+    """Loads the audio data from a NumPy array.
+
+    Args:
+      src: A NumPy source array contains the input audio.
+      offset: An optional offset for loading a slice of the `src` array to the
+        buffer.
+      size: An optional size parameter denoting the number of samples to load
+        from the `src` array.
+    Raises:
+      ValueError: Raised if the input array has an incorrect shape or if
+      `offset` + `size` exceeds the length of the `src` array.
+    """
+    if src.shape[1] != self._format.channels:
+      raise ValueError(
+        f"Input audio contains an invalid number of channels. "
+        f"Expect {self._format.channels}.")
+
+    if size is None:
+      size = len(src)
+    if offset is None:
+      offset = 0
+
+    if offset + size > len(src):
+      raise ValueError(
+        f"Index out of range. offset {offset} + size {size} should be <= "
+        f"src's length: {len(src)}")
+
+    if len(src) >= len(self._buffer):
+      # Copy values from the source array to the internal buffer.
+      new_offset = offset + size - len(self._buffer)
+      new_size = len(self._buffer)
+      self._buffer = src[new_offset:new_offset + new_size].copy()
+    else:
+      # Shift the internal buffer backward and add the incoming data to the end
+      # of the buffer.
+      shift = len(src)
+      self._buffer = np.roll(self._buffer, -shift, axis=0)
```

Comment:
There's a bug that you don't consider the offset here. I fixed it and also updated the test so that it can identify this bug if it's reintroduced. Please pull the latest code from master before continue the development.